modin/engines/ray/utils.py

```diff
@@ -136,7 +136,11 @@ def initialize_ray(
             # In case anything failed above, we can still improve the memory for Modin.
             if object_store_memory is None:
                 # Round down to the nearest Gigabyte.
-                system_memory = ray._private.utils.get_system_memory()
+                system_memory = (
+                    ray._private.utils.get_shared_memory_bytes()
```

Comment:
Previously there was some discussion on #3132 about removing dependency on `ray._private`.

Should we manually check `/dev/shm` size ourselves instead of using this utility?

Reply:
I agree, we should refuse to use private API of Ray in my opinion. We can set `system_memory` by manually checking /dev/shm. But I have a pair of questions:
1) `ray._private.utils.get_shared_memory_bytes()` gives size of /dev/shm (it is half of RAM by default) on Linux. But `ray._private.utils.get_system_memory()` gives size of full RAM on Windows. What size should we use on both OS?
2) Why do we use 0.6 of `system_memory` for `object_store_memory`?

Reply:
>What size should we use on both OS?

On Windows and Mac, we should use the full RAM and on Linux we should use size of /dev/shm. In Windows and Mac they use a temporary file for Plasma, and on Linux they use /dev/shm.

>Why do we use 0.6 of system_memory for object_store_memory?

We need to reserve memory for the workers.

Reply:
> On Windows and Mac, we should use the full RAM and on Linux we should use size of /dev/shm. In Windows and Mac they use a temporary file for Plasma, and on Linux they use /dev/shm.

Thanks, I understood.

> We need to reserve memory for the workers.

I understand this, thanks. But why are we using *exactly* 0.6?

Reply:
>But why are we using exactly 0.6?

It seemed rational, but there really isn't a good reason. Do you have ideas on how it can be improved?

Reply:
Unfortunately, I haven't concrete ideas related this right now.

Reply:
Dependency on `ray._private` was removed